Code/GraphMol/AddHs.cpp

```diff
@@ -1078,6 +1078,22 @@ enum class HydrogenType {
   QueryHydrogen
 };
 
+template<class Q>
+std::pair<bool,bool> queryHasHs(Q queryAtom, bool inor=false) {
+  for(auto childit = queryAtom->beginChildren(); childit != queryAtom->endChildren(); ++childit) {
```

Comment:
This should be a bit faster as we don't need to copy anymore.  Let me know if we need some docs on the decision tree here, it's mostly the same as before, except I recurse into Ors now, otherwise we miss [#1,#6,#7]. which is a tree.